### DISCUSSION TITLE
Debugging Issac Sim File Takes Too Much Time for Vscode to Show the Variable Representation

### DISCUSSION THREAD

**Discussion by zoctipus** [NVlabs/curobo#66] [Software Q&A] [answered]
It seems whenever breakpoint is set to MotionGenConfig, or MotionGen constructior, the vscode will take 30 seconds ~ or even more to load the returned value representation and variable into the debug console. And ever step next will take the same amount of time and making debugging, value inspecting process almost impossible.

I wonder what might be a good way to approach variables inspecting process in developing with curobo, and how do you guys debug a specific code?

**Comment by balakumar-s** [comment, marked as answer]
We do not have isaac sim in the loop of debugging. We implement python scripts (no isaac sim here) that reproduce errors/bugs, then debug issues with vscode debug breakpoints. 

We also have a motion generation dump script that will record iterations of optimization (in IK and Trajectory optimization) and dump them as animated usd files. These usd files can be opened in isaac sim or any usd viewer. An example visualization of this dump file is shown here: https://youtu.be/Ux9tFBRLR-A?si=hih-wqWbsJ4tbVEu&t=241. We are in the process of adding an example of this logging mechanism in the next release (this week).
